crates/pixi_manifest/src/manifests/project.rs

```diff
@@ -285,47 +299,72 @@ impl ManifestSource {
     ) -> Result<(), TomlError> {
-        match self {
-            ManifestSource::PyProjectToml(_) => {
-                // Pypi dependencies can be stored in different places
-                // so we remove any potential dependency of the same name before adding it back
-                self.remove_pypi_dependency(
-                    &PyPiPackageName::from_normalized(requirement.name.clone()),
-                    platform,
-                    feature_name,
-                )?;
-                if let FeatureName::Named(name) = feature_name {
-                    self.manifest()
-                        .get_or_insert_toml_array("project.optional-dependencies", name)?
-                        .push(requirement.to_string())
-                } else {
-                    self.manifest()
-                        .get_or_insert_toml_array("project", "dependencies")?
-                        .push(requirement.to_string())
-                }
-            }
-            ManifestSource::PixiToml(_) => {
-                let mut pypi_requirement =
-                    PyPiRequirement::try_from(requirement.clone()).map_err(Box::new)?;
-                if let Some(editable) = editable {
-                    pypi_requirement.set_editable(editable);
-                }
+        // Pypi dependencies can be stored in different places in pyproject.toml manifests
+        // so we remove any potential dependency of the same name before adding it back
+        if matches!(self, ManifestSource::PyProjectToml(_)) {
+            self.remove_pypi_dependency(
+                &PyPiPackageName::from_normalized(requirement.name.clone()),
+                platform,
+                feature_name,
+            )?;
+        }
 
-                let dependency_table = TableName::new()
-                    .with_prefix(self.table_prefix())
-                    .with_platform(platform.as_ref())
-                    .with_feature_name(Some(feature_name))
-                    .with_table(Some(consts::PYPI_DEPENDENCIES));
-
-                self.manifest()
-                    .get_or_insert_nested_table(dependency_table.to_string().as_str())?
-                    .insert(
-                        requirement.name.as_ref(),
-                        Item::Value(pypi_requirement.into()),
-                    );
+        // The '[pypi-dependencies]' or '[tool.pixi.pypi-dependencies]' table is selected
+        //  - For 'pixi.toml' manifests where it is the only choice
+        //  - When explicitly requested
+        //  - When a specific platform is requested, as markers are not supported (https://github.com/prefix-dev/pixi/issues/2149)
+        //  - When an editable install is requested
+        if matches!(self, ManifestSource::PixiToml(_))
+            || matches!(location, Some(PypiDependencyLocation::Pixi))
+            || platform.is_some()
+            || editable.is_some_and(|e| e)
+        {
+            let mut pypi_requirement =
```

Comment:
Do we want to emit a warning if the dependency is added to the `[tool.pixi.pypi-dependencies]` table because it is editable or platform-specific? or no need?

Reply:
This would be resolved by outputting the location.